app/controllers/admin/roles_controller.rb

```diff
@@ -8,22 +8,40 @@ class RolesController < Admin::BaseController
 
     def index
       @roles = Role.where(resource: @conference)
+      tracks = @conference.program.tracks.where.not(submitter: nil)
+      @roles += Role.where(resource: tracks)
       authorize! :index, @role
     end
 
     def show
+      @url = if @track
+               toggle_user_track_admin_conference_role_path(@conference.short_title, @role.name, @track.name.tr(' ', '_'))
```

Comment:
Actually, Rails already has a way of turning a model into the appropriate URL parameter, via [`to_param`](http://api.rubyonrails.org/v4.2.5/classes/ActiveModel/Conversion.html#method-i-to_param).

By implementing `#to_param` like:
```ruby
def to_param
  short_title # or `name` in Role
end
```

you can then just do:

```ruby
toggle_user_track_admin_conference_role_path(@conference, @role, @track)
```